docs/reference/structure-extensions.adoc

```diff
@@ -0,0 +1,137 @@
+:m3: https://lionweb-io.github.io/specification/metametamodel/metametamodel.html
+:mpshelp: https://www.jetbrains.com/help/mps
+:underbracket: &#9141;
+
+= Reference for LionWeb Extensions on MPS' Language Structure Aspect
+:experimental:
+
+LionWeb can represent MPS languages in LionCore.
+It supports most MPS languages without changes to the language.
+
+== Prerequisites
+We amend MPS' language structure aspect with node attributes to support advanced LionWeb features.
+
+To use advanced LionWeb features, add `io.lionweb.mps.structure.attribute` to the _Used Languages_ of the structure aspect of your language.
+
+
+== Concepts as partitions
+In LionWeb, link:{m3}#partitions[partitions] serve as the container for one node tree (similar to _models_ in MPS).
+
+Use intention menu:Toggle LionWeb Partition[] to add a boolean attribute `LionWeb partition` to a concept.
+If no such attribute exists, we assume default value `false`.
+
+
+== Optional properties
+LionWeb supports link:{m3}#Feature.optional[optional properties] on structure level, whereas MPS only considers them on link:{mpshelp}/editor.html#propertycell[editor level].
```

Comment:
But MPS _does_ have it implicitly on containments and references through the `[0..*]` cardinality, right? So, the exporter could derive optionality from the cardinality. Or did you run into a situation where you (would have) needed to amend the cardinality to get a usable result?

Reply:
For references and containments we of course use the information from MPS; only for properties we have to define our own.